opal/mca/btl/tcp/btl_tcp_component.c

```diff
@@ -735,7 +735,7 @@ static char **split_and_resolve(char **orig_str, char *name, bool reqd)
                                         "btl: tcp: Found match: %s (%s)",
                                         opal_net_get_hostname((struct sockaddr*) &if_inaddr),
                                         if_name);
-                    opal_argv_append(&interface_count, &interfaces, strdup(if_name));
```

Comment:
Fixes Coverity 1494435 - the `if_name` is a stack variable for this function. The problem was that the `strdup` of the value was lost because `opal_argv_append` also does a `strdup`.